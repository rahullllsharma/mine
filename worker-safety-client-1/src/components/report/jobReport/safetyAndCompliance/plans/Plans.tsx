import type { DailyReportInputs } from "@/types/report/DailyReportInputs";
import type { SafetyAndComplianceProps } from "@/container/report/safetyAndCompliance/SafetyAndCompliance";
import { Controller, useFormContext } from "react-hook-form";
import ReportSectionHeader from "@/components/layout/report/reportSectionHeader/ReportSectionHeader";
import ReportSectionBlock from "@/components/layout/report/reportSectionBlock/ReportSectionBlock";
import ReportSectionWrapper from "@/components/layout/report/reportSectionWrapper/ReportSectionWrapper";
import FieldRadioGroup from "@/components/shared/field/fieldRadioGroup/FieldRadioGroup";
import { yesOrNoRadioGroupOptions } from "@/radioGroupConstants";
import { getDefaultOption } from "../utils";

export type PlansInputs = {
  comprehensivePHAConducted: string;
  membersReviewedAndSignedOff: string;
};

const plansFormInputPrefix = "safetyAndCompliance.plans";

export default function Plans({
  isCompleted,
}: SafetyAndComplianceProps): JSX.Element {
  const { getValues } =
    useFormContext<Pick<DailyReportInputs, "safetyAndCompliance">>();

  return (
    <ReportSectionWrapper>
      <ReportSectionHeader title="Plans" />

      <ReportSectionBlock>
        <Controller
          name={`${plansFormInputPrefix}.comprehensivePHAConducted`}
          render={({ field: { onChange } }) => (
            <FieldRadioGroup
              label="Was a comprehensive PHA conducted?"
              options={yesOrNoRadioGroupOptions}
              defaultOption={getDefaultOption(
                yesOrNoRadioGroupOptions,
                getValues(`${plansFormInputPrefix}.comprehensivePHAConducted`)
              )}
              onSelect={onChange}
              readOnly={isCompleted}
            />
          )}
        />
      </ReportSectionBlock>

      <ReportSectionBlock>
        <Controller
          name={`${plansFormInputPrefix}.membersReviewedAndSignedOff`}
          render={({ field: { onChange } }) => (
            <FieldRadioGroup
              label="Have all crew members reviewed and signed off on the HASP?"
              options={yesOrNoRadioGroupOptions}
              defaultOption={getDefaultOption(
                yesOrNoRadioGroupOptions,
                getValues(`${plansFormInputPrefix}.membersReviewedAndSignedOff`)
              )}
              onSelect={onChange}
              readOnly={isCompleted}
            />
          )}
        />
      </ReportSectionBlock>
    </ReportSectionWrapper>
  );
}
