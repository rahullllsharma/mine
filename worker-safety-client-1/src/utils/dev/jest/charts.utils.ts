import type { RiskLevelByDate } from "@/components/charts/dailyRiskHeatmap/types";
import { act } from "@testing-library/react";
import { RiskLevel } from "@/components/riskBadge/RiskLevel";

export function chartRenderHelper(): () => void {
  // set up a listener so we can force the width/height
  // this is required to get the chart to render in its ResponsiveWrapper
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let listener: ((rect: any) => void) | undefined;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (window as any).ResizeObserver = class ResizeObserver {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    constructor(ls: any) {
      listener = ls;
    }
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    observe() {}
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    disconnect() {}
  };

  function triggerResize(): void {
    act(() => {
      listener &&
        listener([
          {
            contentRect: {
              x: 0,
              y: 0,
              width: 800,
              height: 800,
              top: 100,
              bottom: 0,
              left: 100,
              right: 0,
            },
          },
        ]);
    });
  }
  return triggerResize;
}

export const sampleHeatmapDates = {
  startDate: "2022-01-29",
  endDate: "2022-02-03",
};

// consumed by the DailyRiskHeatmap
export const sampleRiskLevelByDate: RiskLevelByDate[] = [
  {
    date: "2022-01-29",
    riskLevel: RiskLevel.HIGH,
  },
  {
    date: "2022-01-30",
    riskLevel: RiskLevel.MEDIUM,
  },
  {
    date: "2022-01-31",
    riskLevel: RiskLevel.UNKNOWN,
  },
  {
    date: "2022-02-01",
    riskLevel: RiskLevel.LOW,
  },
  {
    date: "2022-02-02",
    riskLevel: RiskLevel.RECALCULATING,
  },
  {
    date: "2022-02-03",
    riskLevel: RiskLevel.UNKNOWN,
  },
];
