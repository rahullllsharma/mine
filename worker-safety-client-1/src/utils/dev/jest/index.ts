export * from "./intl.utils";
export * from "./testing.utils";
export * from "./charts.utils";
