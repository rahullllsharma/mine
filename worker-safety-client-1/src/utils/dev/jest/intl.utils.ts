/** Mock the default locale when using the Intl.DateTimeFormat class */
export function mockIntlDateTimeFormatLocale(locale: string): void {
  const { DateTimeFormat } = Intl;
  jest
    .spyOn(global.Intl, "DateTimeFormat")
    .mockImplementation((_, options) => new DateTimeFormat(locale, options));
}
