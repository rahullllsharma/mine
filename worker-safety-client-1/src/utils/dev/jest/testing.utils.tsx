/* istanbul ignore file */
import type { ReactNode } from "react";
import type { PropsWithClassName } from "@/types/Generic";
import { fireEvent, render, screen } from "@testing-library/react";
import { FormProvider, useForm } from "react-hook-form";
import { useTenantStore } from "@/store/tenant/useTenantStore.store";
import { TenantMock } from "@/store/tenant/utils/tenantMock";

export const formTemplate = (
  component: ReactNode,
  defaultValues: { [key: string]: any } = {}
) => {
  function Wrapper({ children }: PropsWithClassName): JSX.Element {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  }
  return <Wrapper>{component}</Wrapper>;
};

export const formRender = (
  component: ReactNode,
  defaultValues: { [key: string]: any } = {}
) => {
  return render(formTemplate(component, defaultValues));
};

// to overcome react-select limitation for testing, of not responding to click events
export const openSelectMenu = (role = "combobox", name?: RegExp): void => {
  const inputElement = screen.getByRole(role, { name });
  fireEvent.keyDown(inputElement, { key: "ArrowDown", code: "ArrowDown" });
};

export const mockTenantStore = (mock = TenantMock) => {
  const { setTenant } = useTenantStore.getState();
  setTenant(mock);
};
