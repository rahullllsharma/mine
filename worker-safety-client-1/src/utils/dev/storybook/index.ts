export * from "./WrapperForm.utils";
export * from "./WrapperModal.utils";
export * from "./WrapperScroll.utils";
