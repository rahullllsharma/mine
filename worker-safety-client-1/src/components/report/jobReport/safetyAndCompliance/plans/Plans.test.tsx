import type { DailyReportInputs } from "@/types/report/DailyReportInputs";
import { screen } from "@testing-library/react";
import { formRender } from "@/utils/dev/jest";
import Plans from "./Plans";

const emptyStateValues: DailyReportInputs = {
  safetyAndCompliance: {
    plans: { comprehensivePHAConducted: "", membersReviewedAndSignedOff: "" },
  },
};

const prefilledValues: DailyReportInputs = {
  safetyAndCompliance: {
    plans: { comprehensivePHAConducted: "0", membersReviewedAndSignedOff: "1" },
  },
};

describe(Plans.name, () => {
  it("should render Plans section component radiogroup", () => {
    formRender(<Plans />);

    const radioGroup = screen.getAllByRole("radiogroup");
    expect(radioGroup).toHaveLength(2);
  });

  it("should display the plans section with no values", () => {
    const { asFragment } = formRender(<Plans />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the plans section with empty state values", () => {
    const { asFragment } = formRender(<Plans />, emptyStateValues);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the plans section with prefilled values", () => {
    const { asFragment } = formRender(<Plans />, prefilledValues);
    expect(asFragment()).toMatchSnapshot();
  });
});
