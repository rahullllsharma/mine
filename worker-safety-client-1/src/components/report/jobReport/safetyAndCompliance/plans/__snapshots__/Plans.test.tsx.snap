// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plans should display the plans section with empty state values 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <div
      class="mb-4 text-neutral-shade-100"
    >
      <h5
        class="font-semibold"
      >
        Plans
      </h5>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Was a comprehensive PHA conducted?
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-21"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-23"
            class="flex items-center"
            id="headlessui-radiogroup-option-22"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-23"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-25"
            class="flex items-center"
            id="headlessui-radiogroup-option-24"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-25"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Have all crew members reviewed and signed off on the HASP?
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-26"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-28"
            class="flex items-center"
            id="headlessui-radiogroup-option-27"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-28"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-30"
            class="flex items-center"
            id="headlessui-radiogroup-option-29"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-30"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Plans should display the plans section with no values 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <div
      class="mb-4 text-neutral-shade-100"
    >
      <h5
        class="font-semibold"
      >
        Plans
      </h5>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Was a comprehensive PHA conducted?
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-11"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-13"
            class="flex items-center"
            id="headlessui-radiogroup-option-12"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-13"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-15"
            class="flex items-center"
            id="headlessui-radiogroup-option-14"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-15"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Have all crew members reviewed and signed off on the HASP?
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-16"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-18"
            class="flex items-center"
            id="headlessui-radiogroup-option-17"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-18"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-20"
            class="flex items-center"
            id="headlessui-radiogroup-option-19"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-20"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Plans should display the plans section with prefilled values 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <div
      class="mb-4 text-neutral-shade-100"
    >
      <h5
        class="font-semibold"
      >
        Plans
      </h5>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Was a comprehensive PHA conducted?
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-31"
          role="radiogroup"
        >
          <div
            aria-checked="true"
            aria-labelledby="headlessui-label-33"
            class="flex items-center"
            id="headlessui-radiogroup-option-32"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            >
              <div
                class="w-2.5 h-2.5 rounded-xl bg-brand-urbint-40 shadow-5"
              />
            </div>
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-33"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-35"
            class="flex items-center"
            id="headlessui-radiogroup-option-34"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-35"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Have all crew members reviewed and signed off on the HASP?
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-36"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-38"
            class="flex items-center"
            id="headlessui-radiogroup-option-37"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-38"
            >
              No
            </label>
          </div>
          <div
            aria-checked="true"
            aria-labelledby="headlessui-label-40"
            class="flex items-center"
            id="headlessui-radiogroup-option-39"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            >
              <div
                class="w-2.5 h-2.5 rounded-xl bg-brand-urbint-40 shadow-5"
              />
            </div>
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-40"
            >
              Yes
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
